### PR TITLE
Add HardwareSerial::updateBaudRate(unsigned long baud) 

### PR DESCRIPTION
**Pull request by Jeroen88**
Add HardwareSerial::updateBaudRate(unsigned long baud) to change the baudrate after ::begin() was called.

### CODE REVIEW

**Comment by Jeroen88** [comment]
@d-a-v or @earlephilhower, any chance this gets merged soon? :)

File: cores/esp8266/HardwareSerial.cpp [hunk 1]
```diff
@@ -133,8 +142,8 @@ unsigned long HardwareSerial::detectBaudrate(time_t timeoutMillis)
     return detectedBaudrate;
 }
 
-size_t HardwareSerial::readBytes(char* buffer, size_t size)
-{
+size_t HardwareSerial::readBytes(char* buffer, size_t size)
```
Comment (earlephilhower) [inline cores/esp8266/HardwareSerial.cpp]: Is this a linefeed change here?  I think this whole file is DOS \r\n formatted, but your patch only has \UNIX n.  For now, could you make sure you match the linefeeds of the rest of the file (yes, I hate DOS linefeeds, too, but we need to be consistent in any particular file).

Comment (Jeroen88) [reply cores/esp8266/HardwareSerial.cpp]: I really do not know. I did not touch these lines. Could you please check it?

Comment (d-a-v) [reply cores/esp8266/HardwareSerial.cpp]: It is indeed a crlf change.
And after checking, this PR restores consistency which is not here in current master.
(The Allman process shall have a try-again session)

**Pull request merged**